### DISCUSSION TITLE
CtkEntry input validation

### DISCUSSION THREAD

**Discussion by lorypelli** [TomSchimansky/CustomTkinter#1243] [General]
Please add only numbers input property

**Comment by ElectricCandlelight** [comment]
Validation must be done by yourself, just like in Tkinter. 

```Python
import customtkinter

root = customtkinter.CTk()

def only_numbers(char):
    return char.isdigit()

validation = root.register(only_numbers)
 
input=customtkinter.CTkEntry(root,validate="key", validatecommand=(validation, '%S'))
input.pack()

root.mainloop()
```

**Comment by biobrad** [comment]
I think that something buggy is going on with validations.

I have been wrestling with getting validation to work for the last 2 days.
Last night, i copied and pasted the code from @ElectricCandlelight into a fresh python file, ran it, and it worked.
An entry box popped up and behaved as it should.

So today, I tried to make a few small amendments to it - i started getting this error:
```
Traceback (most recent call last):
  File "C:\D_Drive\programming files\tkinter\plop.py", line 10, in <module>
    input=customtkinter.CTkEntry(root,validate="key", validatecommand=(validation, '%S'))
  File "C:\Users\bradh\miniconda3\envs\tkinter\lib\site-packages\customtkinter\windows\widgets\ctk_entry.py", line 95, in __init__
    check_kwargs_empty(kwargs, raise_error=True)
  File "C:\Users\bradh\miniconda3\envs\tkinter\lib\site-packages\customtkinter\windows\widgets\utility\utility_functions.py", line 18, in check_kwargs_empty
    raise ValueError(f"{list(kwargs_dict.keys())} are not supported arguments. Look at the documentation for supported arguments.")
ValueError: ['validatecommand'] are not supported arguments. Look at the documentation for supported arguments.
```
So I started rolling back the the small changes to see where I went wrong.
I kept getting the error message.

So I rolled it all the way back so the file is just the original script from @ElectricCandlelight and i am still getting the value error command.

![image](https://user-images.githubusercontent.com/55652506/227646963-5199c22c-06bc-4afb-86d7-e9429a76d5ab.png)

Any help would be awesome. Tearing my hair out a bit. 😓
